src/sentry/event_consumer.py

```diff
@@ -38,3 +47,20 @@ def process_event_from_kafka(message):
 
     return process_event(event_manager, project, key,
                          remote_addr, helper, attachments=None)
+
+
+class EventConsumerWorker(AbstractBatchWorker):
+    def process_message(self, message):
+        value = json.loads(message.value())
+        if value.get('should_process', False):
+            return value
+
+    def flush_batch(self, batch):
+        for event in batch:
+            try:
+                process_event_from_kafka(event)
+            except Exception:
+                logger.exception('Error processing event.')
```

Comment:
What goes in `process_message` vs `flush_batch` is a little hard to decide at this point in time. See the batching consumer code for an understanding:

https://github.com/getsentry/batching-kafka-consumer/blob/2596d2216b889197392aa9bec551f01d473adcc4/batching_kafka_consumer/__init__.py#L19-L82

So the thing you should notice right away is that none of our store code is (currently?) setup to operate on batches. *But* that doesn't mean we want to commit Kafka offsets for every message we process, nor do we want to use the dreaded auto-commit (aka "shrug") IMO.

So what I have here currently at least deserializes the data in `process_message`, and then when a batch-threshold is reached we actually iterate the results, writing (serially) to TSDB/Redis/Celery/whatever-else-StoreView-currently-touches.

This does mean the worker will operate in spurts, but I'm not sure it matters?

We could move `process_event_from_kafka` to `process_message`, and only use `flush_batch` to handle the Kafka commit. I think either way makes about as much sense with the code we have today.

Reply:
Shower thought: maybe we don't want to commit to Kafka every 500-1000ms, and I doubt we want to batch up unprocessed events for longer than that (creating even worse "spurts" of work).

So maybe I _should_ move the actual processing code into `process_message`, returning nothing but `True` and lean on `flush_batch` _solely_ for committing to Kafka every ~5 (10, 30, 60?) seconds. In the worst case a crash consumer means event are reprocessed, but we do de-dupe checking in this code path. And to repeat again: baby steps, in the future maybe we can refactor this code to actually operate on batches better, but that's not a task for today IMO.